src/extensions/AdditiveColormap3DExtensions/BaseExtension.js

```diff
@@ -0,0 +1,66 @@
+import { LayerExtension } from '@deck.gl/core';
+import additiveColormap from './additive-colormap-module';
+import { COLORMAPS } from '../../constants';
+
+const defaultProps = {
+  colormap: { type: 'string', value: 'viridis', compare: true }
+};
+
+function removeExtraColormapFunctionsFromShader(colormap) {
+  // Always include viridis so shaders compile,
+  // but otherwise we discard all other colormaps via a regex.
+  // With all the colormaps, the shaders were too large
+  // and crashed our computers when we loaded volumes too large.
+  const discardColormaps = COLORMAPS.filter(
+    i => i !== (colormap || 'viridis')
+  ).map(i => i.replace(/-/g, '_'));
+  const discardRegex = new RegExp(
+    `vec4 (${discardColormaps.join(
+      '(_([0-9]*))?|'
+    )})\\(float x_[0-9]+\\){([^}]+)}`,
+    'g'
+  );
```

Comment:
Can you add a link to something like https://regex101.com/ showing exactly what this regex finds?

Reply:
This just feels pretty flimsy, and I wonder if writing a build-step extension would make more sense. E.g. one JS module per colormap, and dynamically importing each js module by name (instead of creating one module and dyanmically removing code with a regex at runtime).

Reply:
I'll think on this...now that we are using extensions this might be feasible (unlike before).

Reply:
Hmmm, I'm liking this the more I think about it, but it'll kinda mean that we have like 50 files, no?  One for each colormap?